### PR TITLE
Improve developer exception HTTP response

### PR DESCRIPTION
**Pull request by tkn98**
Show all exceptions incl. previous ones and also their types.

Do a summary first, listing all the details then.

Previously only the message and trace of the last exception has been shown.

Additionally add the previous exception (if any) in case a Block could not 
be instantiated.

Todo:
- [x] Wait for first travis build to finish
- [x] Fix typos in commit message
- [x] Wait for feedback
- [x] Wait for second, third travis build to finish
- [x] Fix tests
- [ ] Wait for restarting of some Travis build tasks
- [ ] Wait for feedback

### CODE REVIEW

**Comment by tkn98** [comment]
An example of the response body is:

```
4 exception(s):
Exception #0 (Magento\Framework\Exception\LocalizedException): Invalid block type: Foo\CategoryImages\Block\SubcategoriesImagesBlock as a string
Exception #1 (ReflectionException): Class Foo\CategoryImages\Service\CategoryImageFactory does not exist
Exception #2 (ReflectionException): Class Foo\CategoryImages\Service\CategoryImageFactory does not exist
Exception #3 (Magento\Framework\Exception\LocalizedException): Source class "\Foo\CategoryImages\Service\CategoryImage" for "Foo\CategoryImages\Service\CategoryImageFactory" generation does not exist.

Exception #0 (Magento\Framework\Exception\LocalizedException): Invalid block type: Foo\CategoryImages\Block\SubcategoriesImagesBlock as a string
#0 /home/tkn98/foo/foo-magento2-shop/shop/src/vendor/magento/framework/View/Layout/Generator/Block.php(237): Magento\Framework\View\Layout\Generator\Block->getBlockInstance('Foo\\CategoryIm...', Array)
#1 /home/tkn98/foo/foo-magento2-shop/shop/src/vendor/magento/framework/View/Layout/Generator/Block.php(216): Magento\Framework\View\Layout\Generator\Block->createBlock('Foo\\CategoryIm...', 'category.subcat...', Array)
#2 /home/tkn98/foo/foo-magento2-shop/shop/src/vendor/magento/framework/View/Layout/Generator/Block.php(124): Magento\Framework\View\Layout\Generator\Block->generateBlock(Object(Magento\Framework\View\Layout\ScheduledStructure), Object(Magento\Framework\View\Layout\Data\Structure), 'category.subcat...')
#3 /home/tkn98/foo/foo-magento2-shop/shop/src/vendor/magento/framework/View/Layout/GeneratorPool.php(86): Magento\Framework\View\Layout\Generator\Block->process(Object(Magento\Framework\View\Layout\Reader\Context), Object(Magento\Framework\View\Layout\Generator\Context))
#4 /home/tkn98/foo/foo-magento2-shop/shop/src/vendor/magento/framework/View/Layout.php(327): Magento\Framework\View\Layout\GeneratorPool->process(Object(Magento\Framework\View\Layout\Reader\Context), Object(Magento\Framework\View\Layout\Generator\Context))
...
```

The HTTP message is shown in the browser with word-wrap (can't simulate that here on Github) so it can be read w/o horizontal scrolling. The exception with the highest number is the most previous and the one with the lowest the most recent one.

This patch helps a lot with debugging in development situations. I can imagine the development situation can be further improved at those places where previous exceptions are not yet provided to new ones.

File: lib/internal/Magento/Framework/App/Http.php [hunk 1]
```diff
@@ -150,7 +150,7 @@ public function catchException(Bootstrap $bootstrap, \Exception $exception)
      */
     private function handleDeveloperMode(Bootstrap $bootstrap, \Exception $exception)
     {
-        if ($bootstrap->isDeveloperMode()) {
+        if (!$bootstrap->isDeveloperMode()) {
```
Comment (cspruiell) [inline lib/internal/Magento/Framework/App/Http.php]: why was this changed? This block should only be executed if we are in developer mode.

**Comment by cspruiell** [comment]
Please check and fix failed Travis builds.

File: lib/internal/Magento/Framework/App/Http.php (hunk 1 above)
Comment (tkn98) [inline lib/internal/Magento/Framework/App/Http.php]: @cspruiell: Most likely a left-over from an inversion (exit early), I'll correct this and update the PR.

**Comment by tkn98** [comment]
@cspruiell: PR updated and fixed incl. test. Some travis builds fail, please restart them, looks like it's non-related to the PR. Ping me if not.

**Comment by okorshenko** [comment]
@tkn98 your Pull Request merged to develop branch. Thank you for contribution!
